CMakeLists.txt

```diff
@@ -23,8 +23,8 @@ include_directories(SYSTEM "${PROJECT_SOURCE_DIR}/third_party/cereal/include")
 include_directories(SYSTEM "${PROJECT_SOURCE_DIR}/third_party/fast-cpp-csv-parser/")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -fexceptions -frtti")
+set(CMAKE_CXX_FLAGS "-std=c++14 -Wshadow -Wswitch-default -Wswitch-enum -Wundef -Wuninitialized -Wpointer-arith -Wcast-align -Wformat=2 -Wredundant-decls -Wreorder ${CMAKE_CXX_FLAGS}")
```

Comment:
can we do -Wall and -Wextra, or is that asking too much?  :)  I think orion also chokes on -Wsign-compare.